search/arch_draft.py

```diff
@@ -0,0 +1,82 @@
+"""
+This is temporary module with the arch concept for the search.
+
+@todo #331:60m  Implement the new arch at the search app.
+ All you need is to implement ContextsStack class
+ and place the other classes to the relevant places.
+ `search.search` module will be fully rewritten.
+ You also can drop away starting_by_term functionality.
+ See "is_name_start_by_term" aggregation parameter
+ inside search.search.search function.
+"""
+
+import abc
+from itertools import chain
+
+from django.db.models import QuerySet
+from django.http import JsonResponse
+
+from images.models import Image
+from pages import typing
+
+
+class Representable(abc.ABC):
```

Comment:
What is this class for?

Reply:
@artemiy312 , it's for clear contract for objects that come as search results by different QuerySets.
I'll add this thought at the pydoc string